parcels/particlefile/particlefileaos.py

```diff
@@ -69,17 +92,17 @@ def read_from_npy(self, file_list, n_timesteps, var):
         :param file_list: List that  contains all file names in the output directory
         :param n_timesteps: Dictionary with (for each particle) number of time steps that were written in out directory
         :param var: name of the variable to read
+        :returns data dictionary of time instances to be written
         """
         max_timesteps = max(n_timesteps.values()) if n_timesteps.keys() else 0
-        data = np.nan * np.zeros((len(n_timesteps), max_timesteps))
-        time_index = np.zeros(len(n_timesteps))
+        data = np.nan * np.zeros((len(n_timesteps), max_timesteps))  # there is the issue that this array should be structured - we need to define its type here!
```

Comment:
OK, so what does this comment mean? Do we need to fix this in this PR?

Reply:
I don't know. Apparently it still works - it's just weird that it works because the `data` of particles are arrays with structures (i.e. arrays where each attribute has a different dataype). Thus, an allocation with `np.zeros` for `var` (where `var` is something different in each call of the function) usually shouldn't work. The fact that it does means it allocates always 64-bit for the variable (because `np.zeros(...)` doesn't mention a `dtype`), meaning we're wasting memory here, at the very least.

So, the comment is true. That said, it works, it has somehow miraculously always worked, so it's not such a pressing issue as to being treated in this PR.